Contains upstream fixes from the ircu repository.

Git commit IDs:

25107247373df4aa75bddc2c2befd7c2dfadec6c

diff -r ed24e1dd118d ChangeLog
--- a/ChangeLog	Sun Aug 11 09:48:30 2013 +0200
+++ b/ChangeLog	Sun Aug 11 09:49:14 2013 +0200
@@ -84,6 +84,11 @@
 	fix a typo in an engine_signal() assertion, and get rid of dead
 	variables in engine_delete().
 
+2010-01-03  Michael Poole <mdpoole@troilus.org>
+
+	* ircd/m_names.c (m_names): Terminate the buffer used for listing
+	clients who are not in any channel.
+
 2010-01-01  Kevin L. Mitchell  <klmitch@mit.edu>
 
 	* doc/readme.xquery: documentation for design and use of the
diff -r ed24e1dd118d ircd/m_names.c
--- a/ircd/m_names.c	Sun Aug 11 09:48:30 2013 +0200
+++ b/ircd/m_names.c	Sun Aug 11 09:49:14 2013 +0200
@@ -288,6 +288,7 @@
 
         if (mlen + idx + NICKLEN + 3 > BUFSIZE)     /* space, \r\n\0 */
         {
+          buf[idx-1] = '\0';
           send_reply(sptr, RPL_NAMREPLY, buf);
           strcpy(buf, "* * :");
           idx = 5;
@@ -295,7 +296,10 @@
         }
       }
       if (flag)
+      {
+        buf[idx-1] = '\0';
         send_reply(sptr, RPL_NAMREPLY, buf);
+      }
       send_reply(sptr, RPL_ENDOFNAMES, "*");
     }
     else if ((chptr = FindChannel(para)) != NULL)
